src/main/java/com/nike/cerberus/jobs/KpiMetricsProcessingJob.java

```diff
@@ -0,0 +1,117 @@
+import org.knowm.sundial.Job;
+import org.knowm.sundial.exceptions.JobInterruptException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Optional;
+
+/**
+ * Periodically send KPI metrics to the enabled metrics services.
+ */
+@Singleton
+public class KpiMetricsProcessingJob extends Job {
+
+    private static final Logger log = LoggerFactory.getLogger(KpiMetricsProcessingJob.class);
+
+    private final CodahaleMetricsCollector metricsCollector;
+
+    private final SafeDepositBoxService safeDepositBoxService;
+
+    private final AwsIamRoleDao awsIamRoleDao;
+
+    private final SecureDataService secureDataService;
+
+    private final UserGroupPermissionService userGroupPermissionService;
+
+    @Inject
+    public KpiMetricsProcessingJob(CodahaleMetricsCollector metricsCollector,
+                                   SafeDepositBoxService safeDepositBoxService,
+                                   AwsIamRoleDao awsIamRoleDao,
+                                   SecureDataService secureDataService,
+                                   UserGroupPermissionService userGroupPermissionService) {
+        this.metricsCollector = metricsCollector;
+        this.safeDepositBoxService = safeDepositBoxService;
+        this.awsIamRoleDao = awsIamRoleDao;
+        this.secureDataService = secureDataService;
+        this.userGroupPermissionService = userGroupPermissionService;
+    }
+
+    @Override
+    public void doRun() throws JobInterruptException {
+        log.debug("Running KPI metrics processing job");
+        try {
+            processKpiMetrics();
+        } catch (JobInterruptException e) {
+            throw e;
+        } catch (Exception e) {
+            log.warn("Error processing KPI metrics", e);
+        }
+    }
+
+    public void processKpiMetrics() {
+        int uniqueIamRoles = awsIamRoleDao.getTotalNumberOfUniqueIamRoles();
+        int uniqueOwnerGroups = userGroupPermissionService.getUniqueSDBOwners().size();
+        int numSDBs = safeDepositBoxService.getTotalNumberOfSafeDepositBoxes();
+        int numDataNodes = secureDataService.getTotalNumberOfDataNodes();
+
+        log.debug("Number of IAM roles: {}, Owner Groups: {}, SDBs: {}, Key/Value Pairs: {}",
+                uniqueIamRoles,
+                uniqueOwnerGroups,
+                numSDBs,
+                numDataNodes);
+
+        getOrCreateSettableGauge("unique_iam_roles").ifPresent((gauge) ->
+                gauge.setValue(uniqueIamRoles));
+
+        getOrCreateSettableGauge("unique_owner_groups").ifPresent((gauge) ->
+                gauge.setValue(uniqueOwnerGroups));
+
+        getOrCreateSettableGauge("number_of_sdbs").ifPresent((gauge) ->
+                gauge.setValue(numSDBs));
+
+        getOrCreateSettableGauge("number_of_data_nodes").ifPresent((gauge) ->
+                gauge.setValue(numDataNodes));
+    }
+
+    private Optional<SettableLongGauge> getOrCreateSettableGauge(String name) {
```

Comment:
is this duplicated code at this point?

May be time to have a metrics service with this method.